### ISSUE TITLE
Does Parallel Sampling work for DDPG?

### ISSUE THREAD

**Issue by Random-Word** [openai/rllab#153]
Your DDPG implementation uses the parallel sampler class, but I can't see how often parameter updates are shared between workers.

Do each of the workers perform parameter updates after each iteration, then periodically pool gradients and update all of their weights, or are parameter updates only shared after each epoch?

Thanks for open sourcing this library. It's very useful :+1:

**Comment by dementrock** [comment]
Actually, I think right now parallel_sampler is only used here to evaluate the policy, not when training it.

**Comment by Random-Word** [comment]
Interesting. When I turn number of workers up I definitely see a proportionate number of cores loaded, and that many environments are created. I had assumed that sampling was being used during training.

In the case of TRPO, are the samples from multiple workers used to populate a shared set of paths for the iteration?

**Comment by dementrock** [comment]
Yes. You are right about TRPO.

**Comment by ViktorM** [comment]
https://arxiv.org/abs/1704.03073 in this paper 2 approaches to improve DDPG are suggested, including parallel, asynchronous variant. But they haven't been added to the rllab yet.

**Comment by Random-Word** [comment]
Yes, that was one of the papers I was replicating. When I saw workers as an
option for DDPG, I thought perhaps I might be able to compare my results to
someone else's implementation of their work.
